client/controllers/events/eventIncidentReports.coffee

```diff
@@ -219,32 +218,52 @@ Template.eventIncidentReports.events
       currentOpen.remove()
     if not closeRow
       $parentRow.addClass('open').after $('<tr>').addClass('details')
-      Blaze.renderWithData Template.incidentReport, @, $('tr.details')[0]
+      Blaze.renderWithData(
+        Template.incidentReport,
+        _.extend(isUserEvent: instance.data.isUserEvent, @),
+        $('tr.details')[0]
+      )
 
   'click .reactive-table tbody tr .edit': (event, instance) ->
-    instanceData = instance.data
-    incident =
-      userEventId: instanceData.userEvent._id
-      edit: true
-      incident: @
-    Modal.show 'incidentModal', incident
+    Modal.show 'incidentModal', incident: @
 
-  'click .reactive-table tbody tr .delete': (event, instance) ->
-    Meteor.call 'removeIncidentFromEvent', @_id, instance.data.userEvent._id, (error, res) =>
+  'click .reactive-table tbody tr .remove': (event, instance) ->
+    Meteor.call 'removeIncidentFromEvent', @_id, instance.data.event._id, (error, res) =>
       if error
         notify('error', error.reason)
         return
       instance.$('tr.details').remove()
       $('.tooltip').remove()
       notify('success', 'Incident report removed from event')
 
+  'click .reactive-table tbody tr .delete': (event, instance) ->
+    deleteSelectedIncidents = =>
+      Meteor.call 'deleteIncidents', [@_id], (error, result) =>
+        if error
+          notify('error', error.reason)
+          return
+        instance.$('tr.details').remove()
+        @$('.tooltip').remove()
+        notify('success', 'Incidents Deleted')
+        Modal.hide('confirmationModal')
+    Modal.show 'confirmationModal',
+      primaryMessage: 'Are you sure you want to completely remove this incident?'
+      secondaryMessage: """
+        Deleting an incident removes it from all events in EIDR-Connect.
+        This action should only be taken if the incident contains incorrect data.
+        You can instead remove it from this event alone by unlinking it.
+        Are you sure you want to delete it?
```

Comment:
This may be a little confusing while on a smart event page because the user can't really unlink an incident without modifying the event details.